File: src/project_mobile/app/src/main/java/com/haidar/mulchoqu/MainActivity.java
package com.haidar.mulchoqu;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.haidar.mulchoqu.activity.KategoriActivity;

public class MainActivity extends AppCompatActivity {

    private Button start;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //start listener
        start = findViewById(R.id.start_b);
        start.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, KategoriActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: src/05.Input_Scroll_and_Data_Binding/app/src/main/java/android/example/aboutme/MainActivity.kt
package android.example.aboutme

import android.content.Context
import android.databinding.DataBindingUtil
import android.example.aboutme.databinding.ActivityMainBinding
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.Button


class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private val myName: MyName = MyName("Haidar Oktafiansyah")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        val button: Button = binding.doneButton
        // Register the onClick listener with the implementation above
        button.setOnClickListener { view ->
            // do something when the button is clicked
            binding.myName = myName
            val editText = binding.nicknameEdit
            val nicknameTextView = binding.nicknameText

            myName.nickname = editText.text.toString()
            editText.visibility = View.GONE
            view.visibility = View.GONE
            nicknameTextView.visibility = View.VISIBLE
            val inputMethodManager = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager

            
        }
    }
}